src/cdk/a11y/key-manager/list-key-manager.ts

```diff
@@ -246,10 +246,12 @@ export class ListKeyManager<T extends ListKeyManagerOption> {
       default:
         // Attempt to use the `event.key` which also maps it to the user's keyboard language,
         // otherwise fall back to resolving alphanumeric characters via the keyCode.
-        if (event.key && event.key.length === 1) {
-          this._letterKeyStream.next(event.key.toLocaleUpperCase());
-        } else if ((keyCode >= A && keyCode <= Z) || (keyCode >= ZERO && keyCode <= NINE)) {
-          this._letterKeyStream.next(String.fromCharCode(keyCode));
+        if (isModifierAllowed) {
```

Comment:
I suspect that shift should be acceptable here (with the intent being a capital letter). It works for the native select, at least.

Reply:
Makes sense. Changed.